Code/Framework/AzCore/AzCore/Math/PackedVector4.h

```diff
@@ -0,0 +1,198 @@
+    using PackedVector4f = PackedVector4<float>;
+    using PackedVector4i = PackedVector4<int32_t>;
+
+
+    //===============================================================
+    // Implementation.
+    //===============================================================
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::PackedVector4()
+    {}
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::PackedVector4(TYPE initValue)
+        : m_x(initValue)
+        , m_y(initValue)
+        , m_z(initValue)
+        , m_w(initValue)
+    {}
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::PackedVector4(TYPE x, TYPE y, TYPE z, TYPE w)
+        : m_x(x)
+        , m_y(y)
+        , m_z(z)
+        , m_w(w)
+    {}
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::PackedVector4(const AZ::Vector4& rhs)
+        : m_x(rhs.GetX())
+        , m_y(rhs.GetY())
+        , m_z(rhs.GetZ())
+        , m_w(rhs.GetW())
+    {}
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::PackedVector4(const TYPE* initData)
+        : m_x(initData[0])
+        , m_y(initData[1])
+        , m_z(initData[2])
+        , m_w(initData[3])
+    {}
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::operator TYPE* ()
+    {
+        return &m_x;
+    }
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::operator const TYPE* () const
+    {
+        return &m_x;
+    }
+
+    template<typename TYPE>
+    PackedVector4<TYPE>::operator AZ::Vector4() const
+    {
+        return AZ::Vector4(m_x, m_y, m_z, m_w);
+    }
+
+    template<typename TYPE>
+    void PackedVector4<TYPE>::Set(TYPE x, TYPE y, TYPE z, TYPE w)
+    {
+        m_x = x;
+        m_y = y;
+        m_z = z;
+        m_w = w;
+    }
+
+    template<typename TYPE>
+    TYPE PackedVector4<TYPE>::GetElement(size_t index) const
+    {
+        AZ_MATH_ASSERT(index < 4, "access beyond bounds of PackedVector4");
+        return reinterpret_cast<const TYPE*>(this)[index];
+    }
+
+    template<typename TYPE>
+    void PackedVector4<TYPE>::SetElement(size_t index, TYPE val)
+    {
+        AZ_MATH_ASSERT(index < 4, "access beyond bounds of PackedVector4");
+        reinterpret_cast<TYPE*>(this)[index] = val;
+    }
+
+    template<typename TYPE>
+    TYPE PackedVector4<TYPE>::GetX() const
+    {
+        return m_x;
+    }
+
+    template<typename TYPE>
+    void PackedVector4<TYPE>::SetX(TYPE v)
+    {
+        m_x = v;
+    }
+
+    template<typename TYPE>
+    TYPE PackedVector4<TYPE>::GetY() const
+    {
+        return m_y;
+    }
+
+    template<typename TYPE>
+    void PackedVector4<TYPE>::SetY(TYPE v)
+    {
+        m_y = v;
+    }
+
+    template<typename TYPE>
+    TYPE PackedVector4<TYPE>::GetZ() const
+    {
+        return m_z;
+    }
+
+
+    template<typename TYPE>
+    void PackedVector4<TYPE>::SetZ(TYPE v)
+    {
+        m_z = v;
+    }
+
+template<typename TYPE>
+    void PackedVector4<TYPE>::SetW(TYPE v)
```

Comment:
It looks like an indent is missing on the "template" line.
```suggestion
    template<typename TYPE>
    void PackedVector4<TYPE>::SetW(TYPE v)
```